ffi/framebuffer_pocketbook.lua

```diff
@@ -9,6 +9,32 @@ local framebuffer = {
 
 }
 
+local function _getPhysicalRect(fb, x, y, w, h)
+    local bb = fb.full_bb or fb.bb
+    w, x = BB.checkBounds(w or bb:getWidth(), x or 0, 0, bb:getWidth(), 0xFFFF)
+    h, y = BB.checkBounds(h or bb:getHeight(), y or 0, 0, bb:getHeight(), 0xFFFF)
+    return bb:getPhysicalRect(x, y, w, h)
+end
+
+local function _updatePartial(fb, x, y, w, h)
+    x, y, w, h = _getPhysicalRect(fb, x, y, w, h)
+
+    fb.debug("refresh: inkview partial", x, y, w, h)
+    inkview.PartialUpdate(x, y, w, h)
+end
+
+local function _updateFull(fb, x, y, w, h)
+    fb.debug("refresh: inkview full", x, y, w, h)
+    inkview.FullUpdate()
+end
```

Comment:
I'd do it here, too.

Full her means the `UPDATE_FULL` mxcfb flag (i.e., flash). It doesn't necessarily imply full-screen ;).

Reply:
@NiLuJe there is nothing to feed it to yet. That is why I didn't add it yet. 

When we need it or you want have it for documentation we can add it ofcourse.

Reply:
I'd do it regardless, otherwise it's either weird or begging for being forgotten and being a source of weird bugs later on ;).

Reply:
Oh, wait. You meant to say that the IV function doesn't take any arguments ;).

That's... a problem, because it doesn't match with our uses of this call, where we do expect it to honor the rect...

Reply:
A minor one, true, but annoying nonetheless ;).

Reply:
@NiLuJe it doesn't take arguments. I think it just updates the Full display. It's the function of PocketBook that results in the best quality result. When asking PocketBook they send me the following explanation:

> The FullUpdate has the highest quality of update but it works only for full screen. If you call the FullUpdate, the full screen will be updated through the black, we do not recommend using it often because these often blinkings can annoy user.
> 
> As for the SoftUpdate – actually, it is the same to Partial update. It is recommended to use it only for b/w area, it has a good quality and doesn’t blink the screen, in addition to it - we recommend using FullUpdate from time to time. But this Soft update will do some ghosting for color areas.
> 
> The PartialUpdateHQ - can be used if there are color elements on the screen to reduce ghosting on them. To use it properly – you should know the coordinates of color area and call this update only for this area. It will do good quality of colored image and will not annoy user as much as the FullUpdate does. Please note, that PartialUdateHQ has sence only for devices with color screens.

Reply:
Yep thus seems to be the limitation of the inview library. I've been running it for some time now but it does seems to work quite nicely so not all is bad ;)